spec/models/user_spec.rb

```diff
@@ -3,37 +3,51 @@
 RSpec.describe User, type: :model do
   describe 'Factory' do
     it 'has a valid factory' do
-      # Testear que el factory definido es válido.
+      expect(build(:user)).to be_valid
     end
   end
 
   describe 'Associations' do
-    # Testear asociaciones (shoulda-matchers).
+    it { should have_many(:posts) }
     # https://github.com/thoughtbot/shoulda-matchers#activerecord-matchers
   end
 
   describe 'Presence validations' do
-    # Testear validaciones de presencia (shoulda-matchers).
+    it { should validate_presence_of(:email) }
+    it { should validate_presence_of(:password) }
+    it { should validate_presence_of(:role) }
     # https://github.com/thoughtbot/shoulda-matchers#activemodel-matchers
   end
 
   describe 'Uniqueness validations' do
-    # Testear validaciones de unicidad (shoulda-matchers).
+    subject { create(:user) }
+    it { should validate_uniqueness_of(:email) }
     # https://github.com/thoughtbot/shoulda-matchers#activemodel-matchers
   end
 
   describe 'Length validations' do
-    # Testear validaciones de longitud (shoulda-matchers).
+    it { should validate_length_of(:password).is_at_least(8) }
     # https://github.com/thoughtbot/shoulda-matchers#activemodel-matchers
   end
 
   describe 'Enumeratives' do
-    # Testear definición de enumerativos (shoulda-matchers).
+    it { should define_enum_for(:role) }
   end
 
-  # Testear métodos de instancia y de clase como para el caso de cualquier
-  # otra clase Ruby (similar a testear ApiLoginManager).
   describe '#valid_password?' do
+    subject { create(:user) }
+    context 'when given value is different from password' do
+      it 'returns false' do
+        expect(subject.valid_password?(SecureRandom.uuid)).to be_falsey
```

Comment:
Está bien, pero sabiendo que el resultado sería directamente `false`, podríamos usar `be false` en lugar de `be_falsey`. Además, el test dice que retorna `false`.